index.js

```diff
@@ -55,9 +68,9 @@ exports.deepScrapeTagPage = function (tag) {
     return new Promise(function (resolve, reject) {
         exports.scrapeTag(tag).then(function (tagPage) {
             return Promise.map(tagPage.medias, function (media, i, len) {
-                return exports.scrapePostCode(media.shortcode).then(function (postPage) {
+                return exports.scrapePostCode(media.node.shortcode).then(function (postPage) {
                     tagPage.medias[i] = postPage;
-                    if ((typeof postPage.location !== "undefined") && postPage.location.has_public_page) {
+                    if (postPage.location && postPage.location.has_public_page) {
```

Comment:
"undefined" check was not enough to filter out "null" locations which resulted in `TypeError: Cannot read property 'has_public_page' of null`